File: c#/_20_FactorialDigitSum.cs
using System;
using System.Numerics;

public class _20_FactorialDigitSum{
	public static BigInteger Euler20(){
		BigInteger fact = 1, sum = 0;
		for (int i = 1; i <= 100; i++){
			fact *= new BigInteger(i);
		}
		while (fact > 0){
			sum += fact % 10;
			fact /= 10;
		}
		return sum;
	}
	
	public static void Main(){
		Console.WriteLine(Euler20());
	}
}

File: java/_20_FactorialDigitSum.java
import java.util.*;
import java.math.BigInteger;

public class _20_FactorialDigitSum{
    
    public static void main(String[] args){
        BigInteger fact = BigInteger.valueOf(100);
        for (int i = 99; i > 0; i--){
            fact = fact.multiply(BigInteger.valueOf(i));
        }
        
        // System.out.println(fact);
        
        int digitSum = 0;
        for (char c : fact.toString().toCharArray()){
            digitSum += Character.getNumericValue(c);
        }
        
        System.out.println("The sum of the digits in 100! is " + digitSum); // 648
    }
}